### ISSUE TITLE
Test Drive: Slow to load start menu

### ISSUE THREAD

**Issue by MrCK1** [PCSX2/pcsx2#2085]
***PCSX2 version:***
v1.5.0-dev-2205

***PCSX2 options:***
MTVU disabled, Defaults, except for certain test cases mentioned below

Loading Time Benchmarks: (best to worst)
- Real PS2 HW: **15 seconds**

- EE Cyclerate -3, Framelimiter DISABLED: **20 seconds** (~380 FPS)

- EE Cyclerate -2, Framelimiter DISABLED: **4+ minutes** (~348 FPS) 

- No speedhacks, Framelimiter DISABLED: **5+ minutes**

- No speedhacks, Framelimiter ENABLED: **10+ minutes**

***Plugins used:***
Defaults, GSdx AVX2

***Description of the issue:***
Test Drive takes a very long time to load before it even reaches the start screen. The real PS2 hardware loads in 15 seconds, but takes much longer to do so in PCSX2. The benchmarks above show that while there are ways to circumvent the issue; the solution isn't the most convenient for end-users or accurate to the real hardware behavior.

[Blockdump from loading to start screen](https://github.com/PCSX2/pcsx2/files/1346918/TestDrive_blockdump.zip): (WARNING: size is 3GB uncompressed)

***How to reproduce the issue:***
Boot game, and wait at loading screen after 3 FMVs

***Last known version to work:***
Never

***PC specifications:***
Intel Core i5-6600K@4.7GHz, GTX 1070, Windows 10 64-bit Home

**Comment by atomic83GitHub** [comment]
I know that TDU Unlimited need VF1 fifo to fix slow booting issue, maybe it have the same issue, did you tried it?

**Comment by MrCK1** [comment]
That's a different game, never worked on this one anyways.

**Comment by DIKH3D** [comment]
![image](https://user-images.githubusercontent.com/34287698/33634908-c2b8a568-da0d-11e7-9fe7-ac7aee9a7e44.png)

**Comment by kozarovv** [comment]
If someone want to look deeper into that. Game (US) loop at 0x308A30 - 0x308AD8 (while pres.bin is in EE memory) for very long time makes loading issue. This should load some presentation movies, which seems to not happen even if game fully load with ee cycle trick. This is probably custom streaming module issue, and also is something that i can't properly fix by myself.

I patched that here, just to make loading fast, since whatever is loaded there, seems to not work anyway: 
 https://forums.pcsx2.net/Thread-Fixing-unplayable-games?pid=609743#pid609743

**Comment by MrCK1** [comment]
Thanks again for the patch. We'll have to look further into this in the future as Dobiestation has this loading issue as well.

**Comment by PSI-Rockin** [comment]
After taking a look at this, I can unfortunately confirm that it is a timing issue. The game is trying to load a movie file off the disc using a custom IOP module, but it keeps spamming the IOP with requests to check the disc status. If the EE is too fast compared to the IOP, the IOP will be overwhelmed by requests and thus take much longer to load the file. This is why lower EE cyclerates speed up loading. On DobieStation, speeding up the IOP also makes the game load much more quickly. On real hardware, the EE would be slower due to bus arbitration (sharing the bus with the DMAC) and cache misses, giving the IOP time to do its work.

The patch works by completely skipping loading the file. Not ideal, but it's the easiest solution here. I don't see any way to accurately emulate this game in PCSX2.

**Comment by kozarovv** [comment]
I still think that movies loaded there not work in PCSX2 at all, but basing on your research i was able to create better patch that not break movie loading. 
I created hook in pnach that jump from 0x308A30 to 0xFFFDC, loop there for 7168 (0x1C00) cycles. Then try to get status on IOP side. This was enough to give IOP more cycles, and make it load almost instant with correct loading movies.
 I posted patch on PCSX2 forum. 

Also, does Dobiestation also have issue with demo mode? On pcsx2 when you leave game in main menu, demo mode will run. But car path is broken similar to Stuntman, and Driv3r.
